ai2thor/controller.py

```diff
@@ -410,6 +411,22 @@ def __init__(
         self.add_depth_noise = add_depth_noise
         self.include_private_scenes = include_private_scenes
         self.x_display = None
+        self.gpu_device = gpu_device
+        cuda_visible_devices = list(
+            map(int, filter(lambda y: y.isdigit(),
+            map(lambda x: x.strip(),
+                os.environ.get('CUDA_VISIBLE_DEVICES', '').split(',')))))
+
+        if self.gpu_device:
+            if (type(self.gpu_device) is not int or self.gpu_device < 0):
+                raise ValueError("Invalid gpu_device: '%s'. gpu_device must be >= 0" % self.gpu_device)
+            elif cuda_visible_devices:
+                if self.gpu_device >= len(cuda_visible_devices):
+                    raise ValueError("Invalid gpu_device: '%s'. gpu_device must less than number of CUDA_VISIBLE_DEVICES: %s" % (self.gpu_device, cuda_visible_devices))
```

Comment:
Is it possible for `cuda_visible_devices` to be a subset of the available devices on a machine? I.e. `cuda_visible_devices=[6,7]`? If so then maybe we want to change `self.gpu_device >= len(cuda_visible_devices)` to `self.gpu_device not in len(cuda_visible_devices)`?